docs/use_cases/rag_cli.md

```diff
@@ -0,0 +1,109 @@
+   ```shell
+   $ llamaindex-cli rag --files "./README.md"
+   ```
+   You can also specify a file glob pattern such as:
+   ```shell
+   llamaindex-cli rag --files "./docs/**/*.rst"
+   ```
+1. **Ask a Question**: You can now start asking questions about any of the documents you'd ingested in the prior step:
+   ```shell
+   $ llamaindex-cli rag --question "What is LlamaIndex?"
+   LlamaIndex is a data framework that helps in ingesting, structuring, and accessing private or domain-specific data for LLM-based applications. It provides tools such as data connectors to ingest data from various sources, data indexes to structure the data, and engines for natural language access to the data. LlamaIndex follows a Retrieval-Augmented Generation (RAG) approach, where it retrieves information from data sources, adds it to the question as context, and then asks the LLM to generate an answer based on the enriched prompt. This approach overcomes the limitations of fine-tuning LLMs and provides a more cost-effective, up-to-date, and trustworthy solution for data augmentation. LlamaIndex is designed for both beginner and advanced users, with a high-level API for easy usage and lower-level APIs for customization and extension.
+   ```
+1. **Open a Chat REPL**: You can even open a chat interface within your terminal! Just run `llamaindex-cli rag --chat` and start asking questions about the files you've ingested.
+
+### Supported File Types
+
+Internally, the `rag` CLI tool uses the [SimpleDirectoryReader](/api/llama_index.readers.SimpleDirectoryReader.rst) to parse the raw files in your local filesystem into strings.
+
+This module has custom readers for a wide variety of file types. Some of those may require that you `pip install` another module that is needed for parsing that particular file type.
+
+If a file type is encountered with a file extension that the `SimpleDirectoryReader` does not have a custom reader for, it will just read the file as a plain text file.
+
+See the next section for information on how to add your own custom file readers + customize other aspects of the CLI tool!
+
+## Customization
+
+The `rag` CLI tool is highly customizable! The tool is powered by stitching together the [`IngestionPipeline`](/module_guides/loading/ingestion_pipeline/root.rst) & [`QueryPipeline`](/module_guides/querying/pipeline/root.rst) modules within the [`RagCLI`](https://github.com/run-llama/llama_index/blob/main/llama_index/command_line/rag.py) module.
+
+To create your own custom rag CLI tool, you can simply create an script that instantiates the `RagCLI` class with a `IngestionPipeline` & `QueryPipeline` that you've configured yourself. From there, you can simply run `rag_cli_instance.cli()` in your script to run the same ingestion and Q&A commands against your own choice of embedding models, LLMs, vector DBs, etc.
+
+Here's some high-level code to show the general setup:
```

Comment:
Just to lower the bar slightly, we can explain that this could go into a file called `rag_cli.py`, and then you can run your CLI app with `python ./rag_cli.py`